core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/MetadataService.java

```diff
@@ -102,15 +110,27 @@ public void removeMetadata( String metadataId ) throws PentahoAccessControlExcep
 
   public List<String> getMetadataDatasourceIds() {
     List<String> metadataIds = new ArrayList<String>();
-    try {
-      sleep( 100 );
-      for ( String id : metadataDomainRepository.getDomainIds() ) {
-        if ( isMetadataDatasource( id ) ) {
-          metadataIds.add( id );
+    Set<String> domainIds = metadataDomainRepository.getDomainIds();
```

Comment:
DRY. Do Repeat Yourself.
This is the same threading logic in `DataSourceWizardService.java#getDSWDatasourceIds()`. Both `DataSourceWizardService` and `MetadataService` extend the class `DatasourceService` so this would be a good candidate to place this code. You can also create a utility.